browser/ui/views/brave_shields/cookie_list_opt_in_browsertest.cc

```diff
@@ -0,0 +1,188 @@
+#include "brave/components/brave_shields/common/brave_shield_constants.h"
+#include "brave/components/brave_shields/common/features.h"
+#include "brave/components/brave_shields/common/pref_names.h"
+#include "chrome/browser/browser_process.h"
+#include "chrome/browser/ui/browser.h"
+#include "chrome/browser/ui/browser_list.h"
+#include "chrome/common/chrome_switches.h"
+#include "chrome/test/base/in_process_browser_test.h"
+#include "chrome/test/base/ui_test_utils.h"
+#include "components/prefs/pref_change_registrar.h"
+#include "components/prefs/pref_service.h"
+#include "content/public/test/browser_test.h"
+#include "content/public/test/browser_test_utils.h"
+#include "url/url_constants.h"
+
+namespace brave_shields {
+
+class CookieListOptInBrowserTest : public InProcessBrowserTest {
+ public:
+  static inline constexpr char kRegionalAdBlockComponentTestId[] =
+      "lfgnenkkneohplacnfabidofpgcdpofm";
+
+  static inline constexpr char kRegionalAdBlockComponentTest64PublicKey[] =
+      "MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAqNcRp37CBZCXe1vbmXyobycPxHyE"
+      "KNIgNl6p0XBBxtcZcQOijpY70GjCRzgCL7m1+FBo4MR3FXLiF2aPn/"
+      "QsUR8t7+zfw3XzBVos4Ssexkqpd4/"
+      "4ciASwTXpbuyFOq4Z5dcgJ1afeT9Zj5bmh4ekLpgJ1NzVwCMhEKk6cmSKIaGVo5EEydtlor2"
+      "nkUJrSFuZA6tYZ++"
+      "4BOfhhCtzrvXTZjg7mTlB6ca21NL4oLwtqvJMtF8ddoumh619BB5wOqxLzntC/"
+      "oWyOxf00V5HDC7e/"
+      "DRj9J8jLRFLd4EQUO4Mk+kG3MNy0ph9cqdw6zFR7a2H3LGkl4ejsifM1mUDuJL0cwIDAQAB";
+
+  CookieListOptInBrowserTest() {
+    scoped_feature_list_.InitWithFeatures(
+        {features::kBraveAdblockCookieListOptIn}, {});
+  }
+
+  ~CookieListOptInBrowserTest() override = default;
+
+ protected:
+  virtual void SetUpLocalState() { InitializeFilterLists(); }
+
+  void PreRunTestOnMainThread() override {
+    CookieListOptInBubbleHost::AllowBubbleInBackgroundForTesting();
+    SetUpLocalState();
+    InProcessBrowserTest::PreRunTestOnMainThread();
+  }
+
+  auto* regional_service_manager() {
+    return g_brave_browser_process->ad_block_service()
+        ->regional_service_manager();
+  }
+
+  content::WebContents* GetBubbleWebContents() {
+    auto* host = CookieListOptInBubbleHost::FromBrowser(browser());
+    return host ? host->GetBubbleWebContentsForTesting() : nullptr;
+  }
+
+ private:
+  void InitializeFilterLists() {
+    std::vector regional_catalog = {adblock::FilterList(
+        kCookieListUuid,
+        "https://secure.fanboy.co.nz/fanboy-cookiemonster_ubo.txt",
+        "Easylist-Cookie List - Filter Obtrusive Cookie Notices", {},
+        "https://forums.lanik.us/", kRegionalAdBlockComponentTestId,
+        kRegionalAdBlockComponentTest64PublicKey,
+        "Removes obtrusive cookie law notices")};
+    regional_service_manager()->SetRegionalCatalog(regional_catalog);
+  }
+
+  base::test::ScopedFeatureList scoped_feature_list_;
+};
+
+IN_PROC_BROWSER_TEST_F(CookieListOptInBrowserTest, EnableFromBubble) {
```

Comment:
for clarity can you comment that FirstRun is false by default for tests?

Reply:
Done.